### ISSUE TITLE
gomplate always seems to produce an extra newline in output to STDERR

### ISSUE THREAD

**Issue by pmorch** [hairyhenderson/gomplate#1074]
```bash
$ touch empty.yaml
$ gomplate -f empty.yaml

$ gomplate -f empty.yaml -o /dev/null 

$ gomplate -f empty.yaml | wc -l     

0
$ gomplate -f empty.yaml 2>/dev/null| wc -l
0
```

I expect that rendering an empty file or to /dev/null does not create an extra line of output. From the last two examples, it looks like the extra line is written to STDERR.

**Comment by hairyhenderson** [comment]
Hi @pmorch, what version of gomplate are you using? There was one fix for #996, which was part of v3.9.0, but then there was also a follow-up fix #1058 which hasn't made it into a release yet.

In theory this should only be a cosmetic issue since it's outputting to stderr and not stdout.

**Comment by zacheryph** [comment]
I get this as of `v3.9.0`

**Comment by zacheryph** [comment]
cosmetic? this is very annoying when using gomplate on dozens of files and seeing newlines every single iteration. and piping `2>/dev/null` is a none option as you lose valid error output then.

I think a lot better explanation of this _feature_ would be needed to better understand the reasoning. It seems rather odd to print a newline (to stderr) _ALL_ the time so the output looks better for a single use case? If people want to print a new line shouldn't they print a new line in their own scripts when/where they want one?

**Comment by zacheryph** [comment]
Also I would say this kind of veers away (and oddly) from standard unix tools. Adding a newline to stderr adds a convenience to the single use case of the input file not having a final newline.

`echo` defaults to printing a new line (to stdout) with a flag to disable.
`printf` does not add newlines at all to out or err.
`cat` does not add newlines at all to out or err.

The one place I could see adding a newline as a benefit, would be when the the `-i` flag is used. It'd probably be pretty safe to assume the user wants a newline added at the end (to stdout.) Other than `echo` I'm not aware of any tool that arbitrary outputs a newline (to stdout or stderr.)

**Comment by hairyhenderson** [comment]
@zacheryph I understand your frustration, and I understand that this doesn't make sense to you. It was originally a workaround to another frustrating output issue, though perhaps not as prevalent.

> Also I would say this kind of veers away (and oddly) from standard unix tools.

gomplate is definitely not a standard UNIX tool. While it is useful in some cases in a pipeline, it's never been a goal to make gomplate strictly conform to the UNIX philosophy.

Since this is already fixed in #1058, I'm going to close this now. @pmorch & @zacheryph, if you try out what's currently in `master` and that doesn't suit your needs, feel free to open a new issue.
